### PR TITLE
Extended packed designation functionality

### PR DESCRIPTION
**Pull request by hhsieh00**
Added functionality for extended packed provisional designations to be implemented by the MPC in anticipation of higher discovery rates from LSST (https://minorplanetcenter.net/mpcops/documentation/provisional-designation-definition/)

### CODE REVIEW

**Comment by mkelley** [comment]
I'll do a detailed review later, but it looks like two of your lines are not covered by the tests:

https://app.codecov.io/gh/NASA-Planetary-Science/sbpy/pull/406/blob/sbpy/data/names.py#L164

Can you add a case that raises IndexError or ValueError?  You can test it like so:

```python
def test_raises_error():
    with pytest.raises(IndexError):
        Names.to_packed("bad_input_here")
```

There are a couple other blocks not in our testing suite, but they don't seem to be related to your edits, so I won't make you cover them.

**Comment by hhsieh00** [comment]
Just a note, I think line 140 (which I didn't write, but just put inside a try-except block; but which is currently flagged as uncovered code) is extraneous, since the "s[6:].isdigit()" condition above it should already prevent anything that looks like "1995 A" (i.e., single letter, no subscript) from being passed through that section.  As such, I think it can be deleted, but just wanted to get a second opinion before doing so.

**Comment by mkelley** [comment]
You're right, except that we can get to that line with a cometary fragment style designation.  I'm seeing two problems. 
 Something like "1234 A-A" returns a value, but should raise an error, right?  Also, lines 129 and 131 test for the hyphen that denotes a fragment.  They test `s[-2]` but fragments could be two characters (73P-BA).  The testing suite doesn't check for two character fragments, but it should.

```
In [3]: Names.to_packed("2024 A-A")
Out[3]: 'K24A00a'
```

They are unrelated, so we should proceed with this PR, and open issues for each of them.

**Comment by hhsieh00** [comment]
Oh yeah, I did notice that multi-letter fragments were not accounted for in the code, but also that they do not seem to be accounted for by the MPC's own documentation (https://www.minorplanetcenter.net/iau/info/PackedDes.html), except by making the 7 character packed designation into an 8 character designation since the fragment is simply recorded in plain-text.  And yeah, 1234 A-A seems like a different issue, so maybe better to treat it with a separate issue/PR?

**Comment by hhsieh00** [comment]
By the way, just for traceability, in working on the code for the extended packed provisional designations, I found that the MPC had also extended its scheme for packed permanent designations beyond 620000, so code for handling extended packed permanent designations is also included in this PR.

**Review by mkelley — CHANGES_REQUESTED** [review]
The updates look good.  Since you tagged it v0.6, we will merge after v0.5 is out.

File: sbpy/data/names.py
Comment (hhsieh00) [reply sbpy/data/names.py]: done!

**Review by mkelley — APPROVED** [review]
You'll need to rebase with the updated main branch, then resolve the changes.rst conflicts.  Let me know if you want some tips.

**Comment by hhsieh00** [comment]
Hm, yeah, can you send me info on how to do that?  Thanks!

**Comment by mkelley** [comment]
Congrats!

**Pull request merged**
